python/ray/tests/test_actor_failures.py

```diff
@@ -1209,6 +1197,72 @@ def ping(self):
     assert " Worker unexpectedly exits" not in str(exc_info.value)
 
 
+def test_actor_restart_and_actor_received_task(shutdown_only):
```

Comment:
This test is similar to https://github.com/ray-project/ray/blob/ad4e989ebffd7a0d4068dd891be79f032082164c/java/test/src/main/java/io/ray/test/ActorRestartTest.java#L90.

All Python CI tests for https://github.com/ray-project/ray/pull/51904/commits/d20880d07e38c4b2009c556f65522ebb2436cbd7 passed, but the Java test mentioned above failed. Hence, I add this test to Python.